src/background.js

```diff
@@ -56,18 +56,8 @@ window.scrNoti.messageOnUpdatedListener = async (
   }
 
   if (changedProperties.read) {
-    // We keep the message id in the seenMessages until we delete the message,
-    // so that the message does not show up again as new
+    seenMessages[message.folder.accountId + message.folder.path].delete(message.id);
```

Comment:
No use of "seenMessages" for the "simple" mode please. Everything else seems fine!

Reply:
In the case of the "simple" mode, `seenMessages` is an empty set, so it boils down, what is more expensive:
1. Do the dictionary look-up and remove an element from an empty set, or
2. Get the configuration for the mode, check that value in an `if` clause and then do 1.?

Reply:
As already discussed, I'm worrying about `seenMessages[message.folder.accountId + message.folder.path]` being `undefined` and calling `.delete()` on it generating an error.

Reply:
Ok.

As you know, the `seenMessages` variable currently is a global variable. How would I make this a class attribute? IMHO this would make everything a lot easier: During initialization of the class, this class attribute would be set to a `Set` or to `null` and then I can use this information later.

Would I just set `window.scrNoti.seenMessages = null;` in the `window.scrNoti.main()` method and then refer to `window.scrNoti.seenMessages` when needed?

Reply:
No need for class. `const seenMessages = {};` already defines properly the `seenMessages` object before any other code runs. It is simply less well "scoped" then if you were using `window.scrNoti.seenMessages = {}`.

In fact the issue is _not_ about `seenMessages`, it's about _its perperties_ (which are objects too): `seenMessages[message.folder.accountId + message.folder.path]`. This code calls the property `message.folder.accountId + message.folder.path` on the (well defined) object `seenMessages`, and, if no property with this key exists on the object `seenMessages`, then  _undefined_ is returned and it is on _that_ `undefined` property object that `.delete()` may be called and may throw an error!

I think everything is fine for the "extended" mode because you call `window.scrNoti.updateSeenMessages()` when it should and this creates the correct properties on the `seenMessages` object. But those properties are _not_ created when the mode is "simple": https://github.com/electrotype/thunderbird-addon-scriptable-notifications/blob/main/src/background.js#L312 . And this is what we want, in my opinion: The simple mode doesn't need to touch the `seenMessages` object at all. That's why I should prevent the line this comment is about to be ran when it's simple mode.

Reply:
I know, it is not about `seenMessages` and I already have accepted your point.

I was thinking at the `seenMessages` variable in general. At the moment, it looks to me to be a global variable (but then please do not forget, that I have no JavaScript experience whatsoever) which I was taught to avoid. But if this is ok with JavaScript, then I will not question or change it!

With the approach outlined above, I also thought, that I could get rid of the asynchronous function calls to `messenger.storage.local.get()` (and to store its result). But this is certainly a minor point.

BTW I enclosed the deletion in an `if` clause, so I hope it is done.